src/v/cluster/tm_stm.cc

```diff
@@ -167,11 +167,59 @@ ss::future<checked<model::term_id, tm_stm::op_status>> tm_stm::do_barrier() {
         });
 }
 
-ss::future<> tm_stm::checkpoint_ongoing_txs() { return ss::now(); }
+ss::future<> tm_stm::checkpoint_ongoing_txs() {
+    if (!use_new_tx_version()) {
+        co_return;
+    }
+    // Wait for all pending state changes to be applied.
+    if (!co_await wait_no_throw(_c->committed_offset(), _sync_timeout)) {
```

Comment:
I am not sure about it. Because now we write data_batches for tx with `ack = leader_ack`
It means committed offset will not be updated.

Reply:
When we use `update_tx` we wait until the changes are applied so they are already there; and mem thing is updated immediately so it's there too.